### PR TITLE
feat: tailwind components lib

### LINKED ISSUE #3192: Possibility to search in SSR catalogue topics tree
**Is your feature request related to a problem? Please describe.**
The topics tree is very extensive, this makes it nearly impossible to find a topic by just clicking through
(https://data-catalogue.molgeniscloud.org/catalogue/ssr-catalogue/all/variables)

**Describe the solution you'd like**
An option to search through the topic tree by text

### PR DESCRIPTION
**Pull request by connoratrug**
Setup separate app to develop, share and test vue components styled using the tailwind css library
This pr initialises the component lib and implements a solution for #3192 as a test case.

The app is pure nuxt, no components library framework is used 

this pr changes the way the theming is done for the ssr-catalogue, previously a css file was generated at build time for each theme based on separate config files, this pr changes this to use a single config file that use ccs variables as config values. The theme is set by passing different values for variables ( at run time ), theme variable values are set in main.css    

Alternatives considered: 
- [vue story](https://storybook.js.org/ ) book ( to many dependancies ( including react layer)
- [histoure](https://histoire.dev/) , ( multiple blocking issues and lack of activity by maintainer, might be worth it to re-evaluate at some point, but bit worried about reliance on single maintainer that has lost of (side)projects)      

Closes #3192

how to test:
- https://preview-emx2-pr-3449.dev.molgenis.org/apps/tailwind-components/
- http://localhost:8080/apps/tailwind-components/

todo:
[ ] fix theme style loading

### CODE REVIEW

File: apps/tailwind-components/components/input/Tree.vue [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        : 'text-search-filter-group-title',
+    ]"
+  >
+    <li v-for="node in nodes" :key="node.name" class="mt-2.5 relative">
+      <span class="flex items-center">
+        <span
+          v-if="node.children?.length"
+          @click.stop="toggleExpand(node.name)"
+          class="-left-[11px] top-0 rounded-full hover:cursor-pointer h-6 w-6 flex items-center justify-center absolute z-20"
+          :class="[
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/input/Tree.vue:96]: I think the use of class can be combined into one and styled using [groups](https://tailwindcss.com/docs/hover-focus-and-other-states#differentiating-nested-groups).

File: apps/tailwind-components/components/input/Tree.vue [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            :width="20"
+            class="text-blue-200 absolute -top-[9px]"
+          />
+        </template>
+      </span>
+      <div class="flex items-start ml-4">
+        <div class="flex items-center">
+          <input
+            type="checkbox"
+            :indeterminate="
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/input/Tree.vue:128]: The indeterminate status seems to stop at the third level.

File: apps/tailwind-components/components/input/Tree.vue [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        >
+          <BaseIcon
+            :name="
+              expandedNodes.includes(node.name) ? 'caret-down' : 'caret-up'
+            "
+            :width="20"
+          />
+        </span>
+        <template v-if="!isRoot">
+          <BaseIcon
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/input/Tree.vue:110]: The button needs to indicate that it controls a nested element. `aria-controls` and `aria-expanded` is needed.

Comment (davidruvolo51) [reply apps/tailwind-components/components/input/Tree.vue:110]: There should also be a label here that describes what the button does

File: apps/tailwind-components/components/input/Tree.vue [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+          class="-left-[11px] top-0 rounded-full hover:cursor-pointer h-6 w-6 flex items-center justify-center absolute z-20"
+          :class="[
+            inverted
+              ? 'text-search-filter-group-toggle-inverted hover:bg-search-filter-group-toggle-inverted'
+              : 'text-search-filter-group-toggle hover:bg-search-filter-group-toggle',
+          ]"
+        >
+          <BaseIcon
+            :name="
+              expandedNodes.includes(node.name) ? 'caret-down' : 'caret-up'
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/input/Tree.vue:104]: Perhaps css could be used here instead? I'm wondering what the performance differences are when inserting/removing elements vs rotating the icon

File: apps/tailwind-components/pages/Modal.story.vue [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+let sectionCount = ref(0);
+</script>
+<template>
+  <button
+    @click="showModal"
+    class="flex items-center border rounded-full h-10.5 px-5 text-heading-lg gap-3 tracking-widest uppercase font-display bg-button-primary text-button-primary border-button-primary hover:bg-button-primary-hover hover:text-button-primary-hover hover:border-button-primary-hover"
+  >
+    Open modal 🙈
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/Modal.story.vue:16]: 🙈 <- nice!

**Review by davidruvolo51 — APPROVED** [review]
This is great! I look forward to working with the components library. 

I have some comments and ideas for future development.

File: apps/catalogue.code-workspace
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/catalogue.code-workspace:1]: added this by mistake , maybe it should be moved to .gitignore ? but we could have it as option for developers using vs code, but yea something to discusses but maybe better if i remove it ( .gitignore) it for this pr

File: apps/nuxt3-ssr/components/global/icons/Search.vue
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/nuxt3-ssr/components/global/icons/Search.vue:1]: probably a good idea , these icons used now we passed on from the design agency ( also used in the figma designs ) there a combination of icons ( from google ) and another lib ( used for more complex images ( for example for networks of database , more image like stuff). As long as it ends up as svg i think were oke , could check if het heroicons style fits with the current design. would suggest to do this as a separate PR

File: apps/tailwind-components/assets/css/main.css [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+@font-face {
+  font-family: "IBM Plex Sans";
+  font-style: italic;
+  src: url("../fonts/IBMPlexSans-Italic.woff2");
+}
+
+@layer components {
+  [type="checkbox"]:checked {
+    background-image: url("data:image/svg+xml,%3csvg viewBox='0 0 16 16' fill='%23262626' xmlns='http://www.w3.org/2000/svg'%3e%3cpath d='M12.207 4.793a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0l-2-2a1 1 0 011.414-1.414L6.5 9.086l4.293-4.293a1 1 0 011.414 0z'/%3e%3c/svg%3e");
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/tailwind-components/assets/css/main.css:37]: :S , probably not an issue with this pr but with the current master ( this pr move the icons ) , i'll check and create an issue , can i see this in the variables harmonisation view ?

File: apps/tailwind-components/components/input/Tree.vue (hunk 4 above)
Comment (connoratrug) [reply apps/tailwind-components/components/input/Tree.vue:104]: yes would probably be faster, think this is also done on other icons , i'll have a look

File: apps/tailwind-components/components/input/Tree.vue (hunk 2 above)
Comment (connoratrug) [reply apps/tailwind-components/components/input/Tree.vue:128]: i'll try and reproduce the issue , if im not able to i'll ask for some steps to reproduce this behaviour

File: apps/tailwind-components/components/input/Tree.vue (hunk 3 above)
Comment (connoratrug) [reply apps/tailwind-components/components/input/Tree.vue:110]: note sure if i understand , line 110 - 121 are non actionable icons ( they draw the tree structure with dotted lines) so i woud guess theses do not need labels or aria-controles, or do you mean the button in the lines above this part

Comment (connoratrug) [reply apps/tailwind-components/components/input/Tree.vue:112]: i think i found what you meant, i added the tags
 ```   
   :aria-expanded="expandedNodes.includes(node.name)"
   :aria-controls="node.name"
```

**Review by mswertz — APPROVED** [review]

**Pull request merged**